libs/api/src/lib/graphql/blocks.ts

```diff
@@ -487,6 +496,269 @@ export const GraphQLPublicTeaser = new GraphQLUnionType({
+    take: {type: GraphQLInt},
+    skip: {type: GraphQLInt},
+    teasers: {
+      type: new GraphQLNonNull(new GraphQLList(GraphQLTeaser)),
+      resolve: createProxyingResolver(
+        async ({filter, skip, take, teaserType}, _, {loaders, prisma}) => {
+          if (teaserType === TeaserType.Article) {
+            const articles = await getArticles(
+              {
+                published: true,
+                tags: filter.tags
+              },
+              ArticleSort.PublishedAt,
+              SortOrder.Descending,
+              undefined,
+              skip,
+              take,
+              prisma.article
+            )
+
+            articles.nodes.forEach(article => loaders.articles.prime(article.id, article))
+
+            return articles.nodes.map(
+              article =>
+                ({
+                  articleID: article.id,
+                  style: TeaserStyle.Default,
+                  type: TeaserType.Article,
+                  imageID: null,
+                  lead: null,
+                  title: null
+                } as ArticleTeaser)
+            )
+          }
+
+          if (teaserType === TeaserType.Page) {
+            const pages = await getPages(
+              {
+                tags: filter.tags
+              },
+              PageSort.PublishedAt,
+              SortOrder.Descending,
+              undefined,
+              skip,
+              take,
+              prisma.page
+            )
+
+            pages.nodes.forEach(page => loaders.pages.prime(page.id, page))
+
+            return pages.nodes.map(
+              page =>
+                ({
+                  pageID: page.id,
+                  style: TeaserStyle.Default,
+                  type: TeaserType.Page,
+                  imageID: null,
+                  lead: null,
+                  title: null
+                } as PageTeaser)
+            )
+          }
+
+          if (teaserType === TeaserType.Event) {
+            const pages = await getEvents(
+              {
+                tags: filter.tags
+              },
+              EventSort.StartsAt,
+              SortOrder.Descending,
+              undefined,
+              skip,
+              take,
+              prisma.event
+            )
+
+            pages.nodes.forEach(event => loaders.eventById.prime(event.id, event))
+
+            return pages.nodes.map(
+              event =>
+                ({
+                  eventID: event.id,
+                  style: TeaserStyle.Default,
+                  type: TeaserType.Event,
+                  imageID: null,
+                  lead: null,
+                  title: null
+                } as EventTeaser)
+            )
+          }
+
+          return []
+        }
+      )
+    }
+  },
+  isTypeOf: createProxyingIsTypeOf(value => {
+    return value.type === BlockType.TeaserList
+  })
+})
+
+export const GraphQLPublicTeaserListBlock = new GraphQLObjectType<TeaserListBlock, Context>({
```

Comment:
Requires a public seperate one because we use the same name for public & private. Which means it will have a naming conflict and doesn't allow that.

Reply:
:( unnecessary doubling of code, hopefully there will be time this year to move everything to nest